### ISSUE TITLE
diff with no cache

### ISSUE THREAD

**Issue by dmpetrov** [iterative/dvc#2998]
```
$ dvc -V
0.77.3+e03136
```

The major job of `dvc diff` is to show if there is a file difference and show checksums. This can be done without actual file content. So, no cache is needed. But it looks like the current implementation requires the file content.

It is okay not to show the size diff if the cache is not presented. But the format needs to be changed - `size is ?` not the best way to show the absence of the sizes.

With cache:
```
$ dvc diff -t small.txt 9340a9bb6d49b073b6
dvc diff from 9340a9b to 2dae4a2

diff for 'small.txt'
-small.txt with md5 5f73f210b6202aa07279cdff6776b64f
+small.txt with md5 b4301ef665d0343e028fcd5821654edc

added file with size 6 Bytes
```

Without cache:
```
$ dvc diff -t small.txt 9340a9bb6d49b073b6
dvc diff from 9340a9b to 2dae4a2

diff for 'small.txt'
-small.txt with md5 5f73f210b6202aa07279cdff6776b64f

size is ?
```

**Comment by jorgeorpinel** [comment]
> ...to show if there is a file difference and show checksums
> It is okay not to show the size diff if the cache is not presented. But the format needs to be changed
> ...it looks like the current implementation requires the file content.

It seems none of this applied for new revision of `dvc diff`, after #3229 (I tested). Should we close this?

**Comment by efiop** [comment]
@jorgeorpinel Thanks for the heads up! It wasn't auto-closed because of the formatting. @mroutis Please note that if you close tickets in bulk, you need to add `fix` prefix to every one of those (thought there might be a better way).

**Comment by mroutis** [comment]
Oh! I see, will properly check it next time.

**Comment by jorgeorpinel** [comment]
Oh I see what happened!

![image](https://user-images.githubusercontent.com/1477535/73521341-60d1b680-43cb-11ea-9e99-ab8f94a9b404.png)

Cool, glad it's figured out now 🙂

**Comment by dmpetrov** [comment]
See #3392

**Comment by dmpetrov** [comment]
It seems like it is impossible to get a proper diff with no cache for data directories 😞

Some workaround is needed. Something like pulling data-dir caches when calculating `dvc diff`.

**Comment by efiop** [comment]
Closing in favor of https://github.com/iterative/dvc/issues/3392
